app/models/components/ability_host.rb

```diff
@@ -63,10 +63,33 @@ def instance_instance_user_hash(*roles)
     alias_method :instance_all_instance_users_hash, :instance_instance_user_hash
   end
 
+  module TimeBoundedHelpers
+    protected
+
+    # Creates an array of hashes, which indicates the currently valid start/end time. Note that
+    # if there are more than one conditions, you are supposed to map the array and add more
```

Comment:
This might be better:
> Returns an array of conditions which will return currently valid rows when ORed together in a database query. Reverse-merge each of these hashes with your conditions to obtain the set of currently valid rows in the table.